shell/browser/native_window_mac.mm

```diff
@@ -1234,7 +1238,7 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
       content::DesktopMediaID::TYPE_WINDOW, GetAcceleratedWidget());
   // c.f.
   // https://source.chromium.org/chromium/chromium/src/+/master:chrome/browser/media/webrtc/native_desktop_media_list.cc;l=372?q=kWindowCaptureMacV2&ss=chromium
-  if (base::FeatureList::IsEnabled(features::kWindowCaptureMacV2)) {
+  if (base::FeatureList::IsEnabled(kWindowCaptureMacV2)) {
```

Comment:
/cc @VerteDinde 

I have tried to keep the change from https://github.com/electron/electron/pull/30507 but upstream never disabled `kWindowCaptureMacV2` only `kDesktopCaptureMacV2`. In the linked PR, we were skipping this code path by disabling the feature flag, so the root issue about incorrect window id generated is in our implementation. Would be good to investigate and remove this flag as follow-up. Thanks!